tests/components/blebox/test_switch.py

```diff
@@ -0,0 +1,386 @@
+    feature_mocks[0].async_update = AsyncMock()
+
+    def turn_on1():
+        feature_mocks[1].is_on = True
+
+    feature_mocks[1].async_turn_on = AsyncMock(side_effect=turn_on1)
+    await hass.services.async_call(
+        "switch", SERVICE_TURN_ON, {"entity_id": entity_ids[1]}, blocking=True,
+    )
+
+    assert hass.states.get(entity_ids[0]).state == STATE_OFF
+    assert hass.states.get(entity_ids[1]).state == STATE_ON
+
+
+async def test_switchbox_d_first_off(switchbox_d, hass, config):
+    """Test turning switch on."""
+
+    feature_mocks, entity_ids = switchbox_d
+
+    def initial_update_any():
+        feature_mocks[0].is_on = True
+        feature_mocks[1].is_on = True
+
+    feature_mocks[0].async_update = AsyncMock(side_effect=initial_update_any)
+    feature_mocks[1].async_update = AsyncMock()
+    await async_setup_entities(hass, config, entity_ids)
+    feature_mocks[0].async_update = AsyncMock()
+
+    def turn_off0():
+        feature_mocks[0].is_on = False
+
+    feature_mocks[0].async_turn_off = AsyncMock(side_effect=turn_off0)
+    await hass.services.async_call(
+        "switch", SERVICE_TURN_OFF, {"entity_id": entity_ids[0]}, blocking=True,
+    )
+
+    assert hass.states.get(entity_ids[0]).state == STATE_OFF
+    assert hass.states.get(entity_ids[1]).state == STATE_ON
+
+
+async def test_switchbox_d_second_off(switchbox_d, hass, config):
+    """Test turning switch on."""
+
+    feature_mocks, entity_ids = switchbox_d
+
+    def initial_update_any():
+        feature_mocks[0].is_on = True
+        feature_mocks[1].is_on = True
+
+    feature_mocks[0].async_update = AsyncMock(side_effect=initial_update_any)
+    feature_mocks[1].async_update = AsyncMock()
+    await async_setup_entities(hass, config, entity_ids)
+    feature_mocks[0].async_update = AsyncMock()
+
+    def turn_off1():
+        feature_mocks[1].is_on = False
+
+    feature_mocks[1].async_turn_off = AsyncMock(side_effect=turn_off1)
+    await hass.services.async_call(
+        "switch", SERVICE_TURN_OFF, {"entity_id": entity_ids[1]}, blocking=True,
+    )
+    assert hass.states.get(entity_ids[0]).state == STATE_ON
+    assert hass.states.get(entity_ids[1]).state == STATE_OFF
+
+
+ALL_SWITCH_FIXTURES = ["switchbox", "switchbox_d"]
+
+
+@pytest.mark.parametrize("feature", ALL_SWITCH_FIXTURES, indirect=["feature"])
+async def test_update_failure(feature, hass, config, caplog):
+    """Test that update failures are logged."""
+
+    caplog.set_level(logging.ERROR)
+
+    feature_mock, entity_id = feature
+
+    if isinstance(feature_mock, list):
+        feature_mock[0].async_update = AsyncMock()
+        feature_mock[1].async_update = AsyncMock()
+        feature_mock = feature_mock[0]
+        entity_id = entity_id[0]
+
+    feature_mock.async_update = AsyncMock(side_effect=blebox_uniapi.error.ClientError)
+    await async_setup_entity(hass, config, entity_id)
+
+    assert f"Updating '{feature_mock.full_name}' failed: " in caplog.text
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5535891)